### ISSUE TITLE
Error with --project-name argument in !autotrain llm command

### ISSUE THREAD

**Issue by BCS23090070** [huggingface/autotrain-advanced#581]
Getting error when running on google collab, telling me that i didn't provide project name but yet i did.

!autotrain llm --train --project_name 'Llama2  testing-model' --model meta-llama/Llama-2-7b-chat-hf --data_path vicgalle/alpaca-gpt4 --text_column text --use_peft --use_int4 --learning_rate 2e-4 --train_batch_size 2 --num_train_epochs 3 --trainer sft --model_max_length 2048 --push_to_hub --repo_id student100/llama2-testing -block_size 2048 > training.log &

usage: autotrain <command> [<args>] llm [-h] [--text_column TEXT_COLUMN]
                                        [--rejected_text_column REJECTED_TEXT_COLUMN]
                                        [--prompt-text-column PROMPT_TEXT_COLUMN]
                                        [--model-ref MODEL_REF] [--warmup_ratio WARMUP_RATIO]
                                        [--optimizer OPTIMIZER] [--scheduler SCHEDULER]
                                        [--weight_decay WEIGHT_DECAY]
                                        [--max_grad_norm MAX_GRAD_NORM] [--add_eos_token]
                                        [--block_size BLOCK_SIZE] [--peft] [--lora_r LORA_R]
                                        [--lora_alpha LORA_ALPHA] [--lora_dropout LORA_DROPOUT]
                                        [--logging_steps LOGGING_STEPS]
                                        [--evaluation_strategy EVALUATION_STRATEGY]
                                        [--save_total_limit SAVE_TOTAL_LIMIT]
                                        [--save_strategy SAVE_STRATEGY] [--auto_find_batch_size]
                                        [--mixed-precision MIXED_PRECISION]
                                        [--quantization QUANTIZATION]
                                        [--model_max_length MODEL_MAX_LENGTH] [--trainer TRAINER]
                                        [--target_modules TARGET_MODULES] [--merge_adapter]
                                        [--use_flash_attention_2] [--dpo-beta DPO_BETA]
                                        [--chat_template CHAT_TEMPLATE] [--padding PADDING]
                                        [--train] [--deploy] [--inference] [--username USERNAME]
                                        [--backend BACKEND] [--token TOKEN] [--repo-id REPO_ID]
                                        [--push-to-hub] --model MODEL --project-name PROJECT_NAME
                                        [--seed SEED] [--epochs EPOCHS]
                                        [--gradient-accumulation GRADIENT_ACCUMULATION]
                                        [--disable_gradient_checkpointing] [--lr LR] [--log LOG]
                                        [--data-path DATA_PATH] [--train-split TRAIN_SPLIT]
                                        [--valid-split VALID_SPLIT] [--batch-size BATCH_SIZE]
autotrain <command> [<args>] llm: error: the following arguments are required: --project-name

**Comment by abhishekkrthakur** [comment]
change it to `--project_name 'Llama2testing-model' `

no space.

**Comment by BCS23090070** [comment]
> change it to `--project_name 'Llama2testing-model' `
> 
> no space.

i have removed the space as you mentioned but i still get the very same result , no idea why its happened.

!autotrain llm --train --project_name 'Llama2testing-model' --model meta-llama/Llama-2-7b-chat-hf --data_path vicgalle/alpaca-gpt4 --text_column text --use_peft --use_int4 --learning_rate 2e-4 --train_batch_size 2 --num_train_epochs 3 --trainer sft --model_max_length 2048 --push_to_hub --repo_id student100/llama2-testing -block_size 2048 > training.log &

usage: autotrain <command> [<args>] llm [-h] [--text_column TEXT_COLUMN]
                                        [--rejected_text_column REJECTED_TEXT_COLUMN]
                                        [--prompt-text-column PROMPT_TEXT_COLUMN]
                                        [--model-ref MODEL_REF] [--warmup_ratio WARMUP_RATIO]
                                        [--optimizer OPTIMIZER] [--scheduler SCHEDULER]
                                        [--weight_decay WEIGHT_DECAY]
                                        [--max_grad_norm MAX_GRAD_NORM] [--add_eos_token]
                                        [--block_size BLOCK_SIZE] [--peft] [--lora_r LORA_R]
                                        [--lora_alpha LORA_ALPHA] [--lora_dropout LORA_DROPOUT]
                                        [--logging_steps LOGGING_STEPS]
                                        [--evaluation_strategy EVALUATION_STRATEGY]
                                        [--save_total_limit SAVE_TOTAL_LIMIT]
                                        [--save_strategy SAVE_STRATEGY] [--auto_find_batch_size]
                                        [--mixed-precision MIXED_PRECISION]
                                        [--quantization QUANTIZATION]
                                        [--model_max_length MODEL_MAX_LENGTH] [--trainer TRAINER]
                                        [--target_modules TARGET_MODULES] [--merge_adapter]
                                        [--use_flash_attention_2] [--dpo-beta DPO_BETA]
                                        [--chat_template CHAT_TEMPLATE] [--padding PADDING]
                                        [--train] [--deploy] [--inference] [--username USERNAME]
                                        [--backend BACKEND] [--token TOKEN] [--repo-id REPO_ID]
                                        [--push-to-hub] --model MODEL --project-name PROJECT_NAME
                                        [--seed SEED] [--epochs EPOCHS]
                                        [--gradient-accumulation GRADIENT_ACCUMULATION]
                                        [--disable_gradient_checkpointing] [--lr LR] [--log LOG]
                                        [--data-path DATA_PATH] [--train-split TRAIN_SPLIT]
                                        [--valid-split VALID_SPLIT] [--batch-size BATCH_SIZE]
autotrain <command> [<args>] llm: error: the following arguments are required: --project-name

**Comment by abhishekkrthakur** [comment]
project-name?

**Comment by BCS23090070** [comment]
--project_name 'Llama2testing-model'

**Comment by abhishekkrthakur** [comment]
see the hyphen :) 
ill fix it so it allows underscore too

**Comment by BCS23090070** [comment]
well the output remains the same 

!autotrain llm --train **--project_name 'Llama2testingmodel'** --model meta-llama/Llama-2-7b-chat-hf --data_path vicgalle/alpaca-gpt4 --text_column text --use_peft --use_int4 --learning_rate 2e-4 --train_batch_size 2 --num_train_epochs 3 --trainer sft --model_max_length 2048 --push_to_hub --repo_id student100/llama2-testing -block_size 2048 > training.log &

**Comment by abhishekkrthakur** [comment]
see the hyphen between project and name please

**Comment by BCS23090070** [comment]
sorry but i don't get it ,is it to removed the hyphen within the project name?

i also tried to removed from the command 

!autotrain llm --train --projectname 'Llama2testingmodel' --model meta-llama/Llama-2-7b-chat-hf --data_path vicgalle/alpaca-gpt4 --text_column text --use_peft --use_int4 --learning_rate 2e-4 --train_batch_size 2 --num_train_epochs 3 --trainer sft --model_max_length 2048 --push_to_hub --repo_id student100/llama2-testing -block_size 2048 > training.log &

**Comment by abhishekkrthakur** [comment]
!autotrain llm --train** --project-name **'Llama2testingmodel' --model meta-llama/Llama-2-7b-chat-hf --data_path vicgalle/alpaca-gpt4 --text_column text --use_peft --use_int4 --learning_rate 2e-4 --train_batch_size 2 --num_train_epochs 3 --trainer sft --model_max_length 2048 --push_to_hub --repo_id student100/llama2-testing -block_size 2048 > training.log &

**Comment by BCS23090070** [comment]
okay different output right now ,but still getting other kind of problems haha

!autotrain llm --train --project-name 'Llama2testingmodel' --model meta-llama/Llama-2-7b-chat-hf --data_path vicgalle/alpaca-gpt4 --text_column text --use_peft --use_int4 --learning_rate 2e-4 --train_batch_size 2 --num_train_epochs 3 --trainer sft --model_max_length 2048 --push_to_hub --repo_id student100/llama2-testing -block_size 2048 > training.log &

usage: autotrain <command> [<args>]
AutoTrain advanced CLI: error: unrecognized arguments: --data_path vicgalle/alpaca-gpt4 --use_peft --use_int4 --learning_rate 2e-4 --train_batch_size 2 --num_train_epochs 3 --push_to_hub --repo_id student100/llama2-testing -block_size 2048

i changed all the underscore from the commands into hyphen 

!autotrain llm --train --project-name 'Llama2testingmodel' --model meta-llama/Llama-2-7b-chat-hf --data-path vicgalle/alpaca-gpt4 --text-column text --use-peft --use-int4 --learning-rate 2e-4 --train-batch-size 2 --num-train-epochs 3 --trainer sft --model_max_length 2048 --push-to-hub --repo-id student100/llama2-testing -block-size 2048 > training.log &

usage: autotrain <command> [<args>]
AutoTrain advanced CLI: error: unrecognized arguments: --use-int4 --learning-rate 2e-4 --num-train-epochs 3 -block-size 2048

**Comment by abhishekkrthakur** [comment]
you can get all arguments using "autotrain llm --help".
im not sure where you have taken the command from but they seem quite off and old version.

**Comment by abhishekkrthakur** [comment]
you can also follow the colab link in readme.
